docs/core/integration_diagnostics.md

```diff
@@ -0,0 +1,50 @@
+---
+title: Integration diagnostics
+sidebar_label: "Diagnostics"
+---
+
+Integrations can provide diagnostics to help the user gather data to aid in troubleshooting. Diagnostics can be provided for config entries but also individually for each device entry.
+
+Users can download config entry diagnostics from the config entry options menu, on the integration page. For device diagnostics, users can download them from the device menu.
```

Comment:
The UI is a bit inconsistent with this by the way. For example:

Withings (only has config entry diagnostics):
<img width="366" alt="image" src="https://github.com/user-attachments/assets/bff379ef-d538-441c-8fc0-664e902c02cd">

Spotify (only has config entry diagnostics, but with config url):
<img width="529" alt="image" src="https://github.com/user-attachments/assets/55fe453a-2a31-4a98-8b91-a0db6274df89">

Both are always shown with the dropdown menu in the config entry overview

Reply:
I did not notice it had so many variations. 
I've tried to clarify it a bit more. Let me know if its ok.